src/components/FilterComponent/FilterSelectList/FilterSelectList.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div :id="id" class="sol-filterselectlist">
+    <Input
+      v-if="hasScrolling"
+      id="filter-searchbar"
+      v-model="searchString"
+      class="filter-searchbar"
+      type="search"
+      :placeholder="placeholderSearch"
+    />
+
+    <div
+      ref="filterSelectlist"
+      :class="['list', { 'non-searchable': !hasScrolling }]"
+    >
+      <div
+        v-for="item in items"
+        :key="item.value"
+        class="list-item"
+        @click.prevent.stop="handleClickSelectItem(item)"
+      >
+        <span>{{ item.name }}</span>
+        <Checkbox :id="`${item.name}-checkbox`" :value="isItemSelected(item)" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Input from '@components/Input/Input';
+import Checkbox from '@components/Checkbox/Checkbox';
+
+export default {
+  name: 'FilterSelectList',
+
+  components: {
+    Input,
+    Checkbox,
+  },
+  props: {
+    id: {
+      type: String,
+      default: 'sol-filterselectlist',
+    },
+    filter: {
+      type: Object,
+      required: true,
+      validator: (obj) => {
+        return obj.items && Array.isArray(obj.items);
+      },
+    },
+    value: {
+      type: Array,
+      default: () => [],
+    },
+    placeholderSearch: {
+      type: String,
+      default: 'Busque aqui',
+    },
+  },
+
+  data() {
+    return {
+      searchString: '',
+      data: [...this.value],
+      interval: null,
+      hasScrolling: false,
+    };
+  },
+  computed: {
+    filterSelectlist() {
+      return this.$refs.filterSelectlist || {};
+    },
+    items() {
+      return this.filter.items.filter((item) => {
+        if (!this.searchString) {
+          return true;
+        }
+
+        return item.name.toLowerCase().includes(
+          this.searchString
+            .normalize('NFD')
+            .replace(/[^\w\d\s]/gu, '')
+            .toLowerCase(),
+        );
+      });
+    },
+  },
+  watch: {
+    value() {
+      this.data = [...this.value];
+    },
+  },
+  mounted() {
+    this.interval = setInterval(() => {
+      if (!this.filterSelectlist) {
+        this.hasScrolling = false;
+        return;
+      }
+
+      if (this.searchString !== '') {
+        return;
+      }
+
+      this.hasScrolling =
+        this.filterSelectlist.scrollHeight > this.filterSelectlist.clientHeight;
+    }, 100);
+  },
+
+  destroyed() {
+    clearInterval(this.interval);
+  },
+  methods: {
+    handleClickSelectItem(item) {
+      this.isItemSelected(item)
+        ? this.deselectItem(item)
+        : this.selectItem(item);
+
+      this.$emit('input', this.data);
+      this.$emit('change', this.data);
+    },
+    isItemSelected(item) {
+      return this.data.includes(item.value);
+    },
+
+    selectItem(item) {
+      this.data.push(item.value);
+    },
+
+    deselectItem(item) {
```

Comment:
talvez o correto seja 
```suggestion
    unselectItem(item) {
```

https://www.quora.com/What-is-the-difference-between-Unselect-or-Deselect#:~:text=The%20difference%20between%20deselect%20and,mark%20from%20a%20tick%20box.